language/bytecode-verifier/src/abstract_state.rs

```diff
@@ -138,7 +138,9 @@ impl AbstractState {
                 );
             }
         }
-        let num_locals = module.signature_at(function_definition.code.locals).len();
+        let num_locals = module
+            .signature_at(function_definition.code.as_ref().unwrap().locals)
```

Comment:
The pattern "as_mut()/as_ref().unwrap()" appears a times in the verifier. They are probably fine as we don't seem to run the abstract interpreter passes on a native function. But still this irks me and I'll see if I can clean up the code a bit and get rid of them.

Reply:
Hmm... probably not worth the hassle. What do you guys think of an `assert!`, `debug_assert!` or some variation of `precondition!`? Is the last one a MIRAI thing?

Reply:
I think we will look into precondintion or similar so I would not worry about this for now